src/components/PatchActionButtons/SetPatchVisibility.tsx

```diff
@@ -0,0 +1,68 @@
+import { useState, useRef } from "react";
+import { useMutation } from "@apollo/client";
+import { MenuItem } from "@leafygreen-ui/menu";
+import { Body } from "@leafygreen-ui/typography";
+import Popconfirm from "components/Popconfirm";
+import { useToastContext } from "context/toast";
+import {
+  SetPatchVisibilityMutation,
+  SetPatchVisibilityMutationVariables,
+} from "gql/generated/types";
+import { SET_PATCH_VISIBILITY } from "gql/mutations";
+
+interface Props {
+  patchId: string;
+  refetchQueries?: string[];
+}
+export const SetPatchVisibility: React.VFC<Props> = ({
+  patchId,
+  refetchQueries = [],
+}) => {
+  const dispatchToast = useToastContext();
+  const [open, setOpen] = useState(false);
+  const menuItemRef = useRef<HTMLDivElement>(null);
+
+  const [setPatchVisibility] = useMutation<
+    SetPatchVisibilityMutation,
+    SetPatchVisibilityMutationVariables
+  >(SET_PATCH_VISIBILITY, {
+    onCompleted: () => {
+      dispatchToast.success(`This patch was successfully hidden`);
+    },
+    onError: (err) => {
+      dispatchToast.error(`Unable to hide this patch: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  return (
+    <>
+      <div ref={menuItemRef}>
+        <MenuItem
+          active={open}
+          data-cy="disable"
+          onClick={() => setOpen(!open)}
+        >
+          Hide Patch
+        </MenuItem>
+      </div>
+      <Popconfirm
+        align="left"
+        onConfirm={() => {
+          setPatchVisibility({
+            variables: { patchIds: [patchId], hidden: true },
+          });
+        }}
+        open={open}
+        refEl={menuItemRef}
+        setOpen={setOpen}
+      >
+        <Body weight="medium">Hide Patch?</Body>
+        <Body>
+          Hiding this patch will prevent it from being displayed in the my
+          patches page. This can not be undone!
```

Comment:
Is it true that it cannot be undone? I feel like that may discourage users from using the functionality, in case they make a mistake... but we could technically reverse it right?

```suggestion
          patches page. This cannot be undone!
```

Reply:
I left the warning since undoing it would require our team to do a thing. There wouldn't be a self-service way to do this. In the future, we could provide a way to toggle patches to be visible or not. But there are no immediate plans for this.

Reply:
The feature is designed in a way where creating that future functionality would be possible with minimal modifications.

Reply:
I wonder if we should change the warning to be like "You will have to ask the Evergreen team to undo this" so it sounds less permanent, but it's up to you!

Reply:
I opened this ticket for a follow-up to add a way to reverse it. https://jira.mongodb.org/browse/EVG-19921 

I would hesitate to include verbiage encouraging users to ask us to do a thing manually. If they want to ask we can do it/ and or encourage us to prioritize EVG-19921